src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class TagHelperAttribute<TValue> where TValue : class
+    {
+        private string _key;
+
+        public TagHelperAttribute()
+        {
+        }
+
+        public TagHelperAttribute([NotNull] string key)
+            : this(key, value: null)
+        {
+        }
+
+        public TagHelperAttribute([NotNull] string key, TValue value)
+        {
+            Key = key;
+            Value = value;
+        }
+
+        public string Key
+        {
+            get
+            {
+                return _key;
+            }
+            set
+            {
+                // Keys aren't allowed to be null.
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(Key));
+                }
+
+                _key = value;
+            }
+        }
+
+        public TValue Value { get; set; }
+
+        public bool Minimized { get; set; }
```

Comment:
also, should tag helper authors even be able to override the user's wishes in this way?  for one thing, it's an odd feature to support (minification is something we've recommended doing as a post-processing step).  for another, these instances can be replaced and therefore don't need to be updated in-place.  all in all, make attribute updates aka replacement possible but not easy and especially not easy to do accidentally.